packages/expo-camera/app.plugin.js

```diff
@@ -0,0 +1,60 @@
+const pkg = require('./package.json');
+const {
+  createRunOncePlugin,
+  withPlugins,
+  AndroidConfig,
+  withProjectBuildGradle,
+} = require('@expo/config-plugins');
+
+// Because we need the package to be added AFTER the React and Google maven packages, we create a new allprojects.
+// It's ok to have multiple allprojects.repositories, so we create a new one since it's cheaper than tokenizing
+// the existing block to find the correct place to insert our camera maven.
+const gradleMaven =
+  'allprojects { repositories { maven { url "$rootDir/../node_modules/expo-camera/android/maven" } } }';
+
+const CAMERA_USAGE = 'Allow $(PRODUCT_NAME) to access your camera';
+const MICROPHONE_USAGE = 'Allow $(PRODUCT_NAME) to access your microphone';
+
+const withAndroidCameraGradle = config => {
+  return withProjectBuildGradle(config, config => {
+    if (config.modResults.language === 'groovy') {
+      config.modResults.contents = setGradleMaven(config.modResults.contents);
+    } else {
+      throw new Error('Cannot add camera maven gradle because the build.gradle is not groovy');
+    }
+    return config;
+  });
+};
+
+function setGradleMaven(buildGradle) {
+  // If this specific line is present, skip.
+  // This also enables users in bare workflow to comment out the line to prevent expo-camera from adding it back.
+  if (buildGradle.includes('expo-camera/android/maven')) {
+    return buildGradle;
+  }
+
+  return buildGradle + `\n${gradleMaven}\n`;
+}
+
+const withCamera = (config, { cameraPermission, microphonePermission } = {}) => {
+  if (!config.ios) config.ios = {};
+  if (!config.ios.infoPlist) config.ios.infoPlist = {};
+  config.ios.infoPlist.NSCameraUsageDescription =
+    cameraPermission || config.ios.infoPlist.NSCameraUsageDescription || CAMERA_USAGE;
+  config.ios.infoPlist.NSMicrophoneUsageDescription =
+    microphonePermission || config.ios.infoPlist.NSMicrophoneUsageDescription || MICROPHONE_USAGE;
```

Comment:
What about using `lodash.set` method or similar? 🤔 
```suggestion
  _.set(config, 'ios.infoPlist.NSCameraUsageDescription', cameraPermission ?? config.ios.infoPlist.NSCameraUsageDescription ?? CAMERA_USAGE);
  _.set(config, 'config.ios.infoPlist.NSMicrophoneUsageDescription', microphonePermission ?? config.ios.infoPlist.NSMicrophoneUsageDescription ?? MICROPHONE_USAGE);
```

Reply:
we should definitely not install lodash for this single purpose.

Reply:
we could potentially build something like `lodash.set` as a utility to make this type of thing easier, but we need to be careful about preserving type safety. we can follow up when we convert these to ts once the appropriate infra is in place